src/components/register.js

```diff
@@ -63,6 +68,16 @@ export default class RegisterComponent extends React.Component {
       return;
     }
 
+    // if (form.username.value && this.unavailable) {
+    //   form.username.setCustomValidity('Username is busy');
+    //   return;
+    // }
+
+    if (this.error) {
+      form.username.setCustomValidity('Internal server error');
```

Comment:
why "internal server error"? it could be network glitch. use exception's message instead